src/Utilities/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAbstractValueKind.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    /// <summary>
+    /// Kind for <see cref="TaintedDataAbstractValue"/>.
+    /// </summary>
+    internal enum TaintedDataAbstractValueKind
+    {
+        /// <summary>
+        /// Dunno.
+        /// </summary>
+        Unknown,
```

Comment:
>Unknown [](start = 8, length = 7)

We have used the term "Undefined" for the bottom values, because majority of our value kinds also have an "Unknown" or "MayBe" value as the top (last) enum value. Consider using "Undefined" just for consistency.

Reply:
Now just using Tainted and Untainted.